xarray/core/datatree.py

```diff
@@ -91,14 +91,35 @@ def _coerce_to_dataset(data: Dataset | DataArray | None) -> Dataset:
     return ds
 
 
-def _check_for_name_collisions(
-    children: Iterable[str], variables: Iterable[Hashable]
+def _collect_data_and_coord_variables(
+    data: Dataset,
+) -> tuple[dict[Hashable, Variable], dict[Hashable, Variable]]:
+    data_variables = {}
+    coord_variables = {}
+    for k, v in data.variables.items():
+        if k in data._coord_names:
+            coord_variables[k] = v
+        else:
+            data_variables[k] = v
+    return data_variables, coord_variables
+
+
+def _check_alignment(
+    node_ds: Dataset,
+    parent_ds: Dataset | None,
+    children: Mapping[Hashable, DataTree],
```

Comment:
```suggestion
def _find_alignable_coordinates(
    node_ds: Dataset,
    parent_ds: Dataset | None,
) -> Coordinates:
```

Then we would store those coordinates on the node at construction time but not use them unless the user calls `.inherit`.